esmvalcore/preprocessor/_io.py

```diff
@@ -130,7 +130,14 @@ def load(
         # warnings.filterwarnings
         # (see https://github.com/SciTools/cf-units/issues/240)
         with suppress_errors():
-            raw_cubes = iris.load_raw(file, callback=_load_callback)
+            # GRIB files need to be loaded with iris.load, otherwise we will
+            # get separate (lat, lon) slices for each time step, pressure
+            # level, etc.
+            grib_formats = ('.grib2', '.grib', '.grb2', '.grb', '.gb2', '.gb')
+            if file.suffix in grib_formats:
+                raw_cubes = iris.load(file, callback=_load_callback)
```

Comment:
Manu, you should tell @trexfeathers and Iris folk to tell ECMWF to list Iris as a viable/recommended GRIB loader, see https://confluence.ecmwf.int/display/CKB/What+are+GRIB+files+and+how+can+I+read+them - one q with regards to Iris stability wrt GRIB files - is this a gods-given one now? If not, I'd argue we should perform a few consistency tests on the loaded cube(s)

Reply:
Since the recent update(s) of iris-grib I didn't have any difficulties reading those ERA5 GRIB files. Since I would not expect that these files will change any time soon, I won't expect any problems with that. We also perform the same extensive CMOR checks on those files as we do for the netCDF files, so I am pretty confident we are fine here.

About reading GRIB files in general - I am not 100% sure if we (or better, iris-grib) support all possible GRIB files now, but I wouldn't know what to do about this here. If we encounter problems at some point in the future, we can fix that there. Again, given that we have extensive CMOR checks in our pipeline, I am fairly sure we will know if there are problems, even if we won't get very obvious errors.

Reply:
Hi folks, I've raised SciTools/iris-grib#515

Iris-grib definitely cannot load all possible GRIB files. As you may have noticed, the GRIB specification, and how [eccodes](https://confluence.ecmwf.int/display/ECC) interprets this, is not as specific as NetCDF. This means it can be hard to know for certain if Iris-grib is doing the right thing, and we typically therefore rely on finding expert users who can tell us.

So the list of loadable templates is limited to those where we could find a user to check our work. And this also means that we might later discover problems with our existing work, but so far that appears to be rare.